example_bt2/btx_filter/callbacks.c

```diff
@@ -0,0 +1,59 @@
+#include <metababel/metababel.h>
+#include <stdbool.h>
+#include <assert.h>
+#include <stdio.h>
+#include <string.h>
+
+struct data_s {
+  const char *names_value;
+};
+
+typedef struct data_s data_t;
+
+void btx_initialize_usr_data(void *btx_handle, void **usr_data) {
+  *usr_data = calloc(1, sizeof(data_t));
+}
+
+void btx_read_params(void *btx_handle, void *usr_data, btx_params_t *usr_params) {
+  ((data_t *)usr_data)->names_value = usr_params->names;
+}
+
+void btx_finalize_usr_data(void *btx_handle, void *usr_data) {
+  data_t *data = (data_t *)usr_data;
+  free(data);
+}
+
+static void btx_filter_condition(void *btx_handle, void *usr_data, const char* event_class_name, bool *matched)
+{
+  data_t *data = (data_t *)usr_data;
+
+  int str_size = strlen(data->names_value);
+  char str[str_size];
+
+  strcpy(str, data->names_value);
+
+  bool ocurrence = false;
+
+  char *ptr = strtok(str, ",");
+  while(ptr != NULL && !ocurrence)
+  {
+    ocurrence = (strcmp(ptr,event_class_name) == 0);
+    ptr = strtok(NULL, ",");
+  }
```

Comment:
We should addapt their code:
```
for (uint64_t i = 0; i < [bt_value_array_get_length](https://babeltrace.org/docs/v2.0/libbabeltrace2/group__api-val.html#gaf46c04faf88e6ebd44db4ba40b3b3fe6)(
            distill_iter->distill->names_value); i++) {
        const char *discard_name = [bt_value_string_get](https://babeltrace.org/docs/v2.0/libbabeltrace2/group__api-val.html#ga7e98e940d278a5b7e9c05081d25c1e28)(
            [bt_value_array_borrow_element_by_index_const](https://babeltrace.org/docs/v2.0/libbabeltrace2/group__api-val.html#ga534d3ba2056438eb0181f4a978379517)(
                distill_iter->distill->names_value, i));
 
        if (strcmp(name, discard_name) == 0) {
            passes = false;
            goto end;
        }
    }
```

The goal is to stay as close as possible as their example.

Reply:
They use array of strings, but we do not support that, so I used string. I can convert a [comma separed string](https://github.com/portfoliocourses/c-example-code/blob/main/split.c) to an array, then replicate their code. But this add more lines of code. What do you think?